### PR TITLE
core: pathfinding: reproduce edge case in a unit test + fix

### LINKED ISSUE #5405: core: pathfinding: the path sometimes doesn't include a required waypoint
### What happened?

Core returned a 500, a waypoint wasn't included in the resulting path

### What did you expect to happen?

_No response_

### How can we reproduce it (as minimally and precisely as possible)?

Run the fuzzer on France, it has also been reproduced in a unit test in the commit that will fix this

### What operating system, browser and environment are you using?

- **OS**: Ubuntu
- **Env**: Local

### OSRD version (top right corner `Account` button > `Informations`)

20d6b64c4fa4abf947d8ee211d4907693fa67a2d

### PR DESCRIPTION
**Pull request by eckter**
Fix https://github.com/osrd-project/osrd/issues/5405

### CODE REVIEW

File: core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingBlocksEndpoint.java [hunk 1]
```diff
@@ -323,9 +323,12 @@ private void assertPathTracksAreComplete(List<RJSDirectionalTrackRange> tracksOn
         while (i < tracksOnPath.size()) {
             var track = tracksOnPath.get(i);
             var trackName = track.trackSectionID;
-            var fullTrack = new ArrayList<>(tracksOnPath.stream()
-                    .filter(trackOnPath -> Objects.equals(trackOnPath.trackSectionID, track.trackSectionID))
-                    .toList());
+            var fullTrack = new ArrayList<RJSDirectionalTrackRange>();
+            for (int j = i; j < tracksOnPath.size(); j++) {
+                if (!tracksOnPath.get(i).trackSectionID.equals(tracksOnPath.get(j).trackSectionID))
```
Comment (Erashin) [inline core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingBlocksEndpoint.java]:
```suggestion
                if (!trackName.equals(tracksOnPath.get(j).trackSectionID))
```

**Review by younesschrifi — APPROVED** [review]
LGTM :)

**Review by Erashin — APPROVED** [review]
LGTM

**Review by multun — APPROVED** [review]
